### PR TITLE
fix(message-input): message input recipe tests

### PR DESCRIPTION
**Pull request by sanjeevdialpad**
# fix(message-input): message input recipe tests

## :hammer_and_wrench: Type Of Change

- [x] Fix
- [ ] Feature
- [ ] Refactoring
- [ ] Documentation

### CODE REVIEW

File: recipes/conversation_view/message_input/message_input.test.js [hunk 1]
```diff
@@ -0,0 +1,241 @@
+import { mount } from '@vue/test-utils';
+import DtRecipeMessageInput from './message_input.vue';
+import { beforeEach, describe } from '@/node_modules/vitest/dist/index';
+
+import {
+  itBehavesLikeEmitsExpectedEvent,
+} from '../../../tests/shared_examples/events';
```
Comment (iropolo) [inline recipes/conversation_view/message_input/message_input.test.js]: use alias `@/` import

File: recipes/conversation_view/message_input/message_input.test.js [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+let imageInputEl;
+let messageInputEl;
+let characterLimitEl;
+let errorNoticeEl;
+
+let imageBtn;
+let emojiPickerBtn;
+let sendBtn;
+
+// let emojiPickerPopover;
```
Comment (iropolo) [inline recipes/conversation_view/message_input/message_input.test.js]: Remove it

File: recipes/conversation_view/message_input/message_input.test.js [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+import { beforeEach, describe } from '@/node_modules/vitest/dist/index';
+
+import {
+  itBehavesLikeEmitsExpectedEvent,
+} from '../../../tests/shared_examples/events';
+
+// Wrappers
+let wrapper;
+let editor;
+// let editorEl;
```
Comment (iropolo) [inline recipes/conversation_view/message_input/message_input.test.js]: Remove it

File: recipes/conversation_view/message_input/message_input.test.js [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+    attachTo: document.body,
+  });
+};
+
+describe('DtMessage tests', () => {
+  // Test Setup
+  beforeAll(() => {
+    global.Range.prototype.getClientRects = vi.fn(() => [{}]);
+    global.Range.prototype.getBoundingClientRect = vi.fn(() => [{}]);
+    global.scrollBy = vi.fn();
```
Comment (iropolo) [inline recipes/conversation_view/message_input/message_input.test.js]: Here you can create
const getClientRectsMock = vi.fn(() => [{}]);
const getBoundingClientRectMock = vi.fn(() => [{}]);
const scrollByMock = vi.fn();

File: recipes/conversation_view/message_input/message_input.test.js [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+  });
+
+  beforeEach(async () => {
+    props = baseProps;
+    imageUploadStub = vi.fn();
```
Comment (iropolo) [inline recipes/conversation_view/message_input/message_input.test.js]: Here you can create
const imageUploadStub = vi.fn();

File: recipes/conversation_view/message_input/message_input.test.js [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+        await wrapper.setProps({
+          showNotice: true,
+          noticeMessage: randoText,
+        });
+        _setChildWrappers();
+      });
+
+      it('should fire notice-close event when closed', async () => {
+        expect(errorNoticeEl.exists()).toBe(true);
+        errorNoticeEl.find('button').trigger('click');
```
Comment (iropolo) [inline recipes/conversation_view/message_input/message_input.test.js]: `await` here

File: recipes/conversation_view/message_input/message_input.test.js [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+    describe('When send button is clicked', () => {
+      beforeEach(async () => {
+        await wrapper.setProps({
+          modelValue: randoText,
+        });
+        _setChildWrappers();
+      });
+      // eslint-disable-next-line vitest/expect-expect
+      it('should fire submit event with the text as payload', () => {
+        sendBtn.trigger('click');
```
Comment (iropolo) [inline recipes/conversation_view/message_input/message_input.test.js]: `async` `await` here

File: recipes/conversation_view/message_input/message_input.test.js [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+    });
+
+    // select-media event
+    describe('When image input is selected', () => {
+      beforeEach(async () => {
+        await imageInputEl.trigger('input');
+      });
+
+      it('should emit select-media event', async () => {
+        expect(wrapper.emitted()).toHaveProperty('select-media');
```
Comment (iropolo) [inline recipes/conversation_view/message_input/message_input.test.js]: no necessary `async` here

**Review by iropolo — CHANGES_REQUESTED** [review]
Looks good, thanks, just some updates on async/await

File: recipes/conversation_view/message_input/message_input.test.js (hunk 4 above)
Comment (sanjeevdialpad) [reply recipes/conversation_view/message_input/message_input.test.js]: but we need to pass the spy to the function somewhere after initiatlization. Are you suggesting we instatiate it outside and pass this here?

File: recipes/conversation_view/message_input/message_input.test.js (hunk 5 above)
Comment (sanjeevdialpad) [reply recipes/conversation_view/message_input/message_input.test.js]: not needed, left over code deleted. thanks!

File: recipes/conversation_view/message_input/message_input.test.js (hunk 4 above)
Comment (sanjeevdialpad) [reply recipes/conversation_view/message_input/message_input.test.js]: i initialized it outside and passed it in this beforeAll if thats ok.

Comment (iropolo) [reply recipes/conversation_view/message_input/message_input.test.js]: Yes, cool just to have it organized like the others 👍🏼

**Review by iropolo — APPROVED** [review]
Thanks!! 🚀

**Pull request merged**